public/app/features/dashboard-scene/scene/layout-responsive-grid/ResponsiveGridLayoutManager.tsx

```diff
@@ -54,17 +62,35 @@ export class ResponsiveGridLayoutManager
     return max;
   }
 
-  public getNextPanelId(): number {
-    return getDashboardSceneFor(this).getNextPanelId();
-  }
-
   public removePanel(panel: VizPanel) {
     const element = panel.parent;
     this.state.layout.setState({ children: this.state.layout.state.children.filter((child) => child !== element) });
   }
 
   public duplicatePanel(panel: VizPanel): void {
-    throw new Error('Method not implemented.');
+    const gridItem = panel.parent;
+    if (!(gridItem instanceof ResponsiveGridItem)) {
+      console.error('Trying to duplicate a panel that is not inside a DashboardGridItem');
+      return;
+    }
+
+    const newPanelId = dashboardSceneGraph.getNextPanelId(this);
+    const grid = this.state.layout;
+
+    const newGridItem = gridItem.clone({
+      key: `grid-item-${newPanelId}`,
```

Comment:
We have a `getGridItemKeyForPanelId(newPanelId)` helper to ensure we always get `grid-item-` prefix

Reply:
nice, updated